yew/src/virtual_dom/vtag.rs

```diff
@@ -340,162 +380,96 @@ impl VTag {
-            Reform::Keep => {}
-            Reform::Before(next_sibling) => {
-                let element = if self.tag == "svg"
-                    || parent
-                        .namespace_uri()
-                        .map_or(false, |ns| ns == SVG_NAMESPACE)
-                {
-                    let namespace = SVG_NAMESPACE;
-                    #[cfg(feature = "web_sys")]
-                    let namespace = Some(namespace);
-                    document()
-                        .create_element_ns(namespace, &self.tag)
-                        .expect("can't create namespaced element for vtag")
-                } else {
-                    document()
-                        .create_element(&self.tag)
-                        .expect("can't create element for vtag")
-                };
-
-                if let Some(next_sibling) = next_sibling {
-                    let next_sibling = &next_sibling;
-                    #[cfg(feature = "web_sys")]
-                    let next_sibling = Some(next_sibling);
-                    parent
-                        .insert_before(&element, next_sibling)
-                        .expect("can't insert tag before next sibling");
-                } else if let Some(next_sibling) = previous_sibling.and_then(|p| p.next_sibling()) {
-                    let next_sibling = &next_sibling;
-                    #[cfg(feature = "web_sys")]
-                    let next_sibling = Some(next_sibling);
-                    parent
-                        .insert_before(&element, next_sibling)
-                        .expect("can't insert tag before next sibling");
-                } else {
-                    #[cfg_attr(
-                        feature = "std_web",
-                        allow(clippy::let_unit_value, unused_variables)
-                    )]
-                    {
-                        let result = parent.append_child(&element);
-                        #[cfg(feature = "web_sys")]
-                        result.expect("can't append node to parent");
-                    }
+                // If the ancestor is a tag of the same type, don't recreate, keep the
+                // old tag and update its attributes and children.
+                VNode::VTag(vtag) if self.tag == vtag.tag => Some(vtag),
+                _ => {
+                    ancestor.detach(parent);
+                    None
                 }
-                self.reference = Some(element);
             }
-        }
-
-        self.apply_diffs(&ancestor);
+        });
 
-        // Every render it removes all listeners and attach it back later
-        // TODO(#943): Compare references of handler to do listeners update better
-        if let Some(ancestor) = ancestor.as_mut() {
-            ancestor.captured.clear();
+        if let Some(ref mut ancestor_tag) = &mut ancestor_tag {
+            // Control the values of inputs and text area elements
+            ancestor_tag.refresh_value();
```

Comment:
Shouldn't you move this after next line and change to this?
```suggestion
            self.refresh_value();
```

Reply:
I'm not sure. I thought we needed to update the ancestors value to compare it against the value we want

Reply:
Indeed, you're right. Can you drop a comment to explain why it's needed?

Reply:
Done